Editor/BFImporter.cs

```diff
@@ -112,7 +115,7 @@ private static Texture2D[] DoImportTextures(FntParse parse, string rootPath, Tex
                 }
 
                 TextureImporter texImporter = AssetImporter.GetAtPath(texPath) as TextureImporter;
-                texImporter.textureType = TextureImporterType.GUI;
+                texImporter.textureType = TextureImporterType.Sprite;
```

Comment:
This can be reverted, although it won't work in our case where we want the texture type to be respected (we need `Sprite` instead of `GUI` font textures).

Given that there's no way to query the `textureType` out of a `Texture` object (e.g. looking up the already stored `mainTexture` from the material when loading the font in the `DoImportBitmapFont()` method), if we leave this logic as-is, every time we rebuild the bitmap font, it will change the texture type.

What do you think about not importing the font texture only if it's not already assigned to the font material?

e.g.
```c#
font.material = material;

if(material.mainTexture == null)
{
    Texture2D[] textures = DoImportTextures(parse, rootPath, fnt);

    material.shader = Shader.Find(textures.Length > 1 ? "BFI/Font" + textures.Length : "UI/Default");

    material.mainTexture = textures[0];

    for (int i = 1; i < textures.Length; i++)
    {
        material.SetTexture("_MainTex" + (i + 1), textures[i]);
    }
}

font.characterInfo = parse.charInfos;
```

Reply:
- get texture type:
~~~
TextureImporter texImporter = AssetImporter.GetAtPath(texPath) as TextureImporter;
var texType = texImporter.textureType;
~~~

- Why use Sprite?
I don’t understand why you have to use Sprite, just because `Editor GUI and Legacy GUI` seems to be deprecated.

Reply:
> I don’t understand why you have to use Sprite, just because Editor GUI and Legacy GUI seems to be deprecated.

I have reviewed the cases where we used a Sprite type in our project and it looks like this is not needed any more. Maybe we had to switch to that type at some point in the past, but it looks like the `.GUI` type works as well.

A minor suggestion would be to just remove the line where the script sets the explicit `textureType` so that it respects the type set by the user, but it's not a dealbreaker right now; just a suggestion.

Reply:
I have reverted the change back to the `GUI` type.

Reply:
Thanks.
I will add an option for the texture type on weekend.